lib/spack/env/cc: tolerate trailing / in elements of $PATH

bernhardkaindl opened the pull request:
Fixes removal of **SPACK_ENV_PATH** from **PATH** in the presence of trailing slashes in the elements of PATH.

From the commit message:

>    The compiler wrapper has to ensure that it is not called nested like
    it would happen when gcc's collect2 uses PATH to call the linker ld,
    or else the compilation fails.
>
>    To prevent nested calls, the compiler wrapper removes the elements
    of SPACK_ENV_PATH from PATH.
>
>    Sadly, the autotest framework appends a slash to each element
    of PATH when adding AUTOTEST_PATH to the PATH for the tests,
    and some tests like those of GNU bison run cc inside the test.
>    
>    Thus, ensure that PATH cleanup works even with trailing slashes.
>    
>    This fixes the autotest suite of bison, compiling hundreds of
    bison-generated test cases in a autotest-generated testsuite.
    
    Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>

haampie commented on lib/spack/env/cc:
```diff
@@ -616,6 +616,9 @@ if [[ $SPACK_TEST_COMMAND == dump-args ]]; then
     exit
+elif [[ $SPACK_TEST_COMMAND =~ dump-env-* ]]; then
+    var=${SPACK_TEST_COMMAND#dump-env-}
+    eval echo "$0: \$var=\${$var}"
```
```suggestion
    echo "$0: $var: ${!var}"
```
you can use this bash trick to avoid eval

bernhardkaindl replied:
Thanks, neat bash trick!

I committed your suggetion and updated the commit message, no other change in the new pushes.

bernhardkaindl committed:
lib/spack/env/cc: tolerate trailing / in elements of $PATH

Fixes removal of SPACK_ENV_PATH from PATH in the presence of trailing
slashes in the elements of PATH:

The compiler wrapper has to ensure that it is not called nested like
it would happen when gcc's collect2 uses PATH to call the linker ld,
or else the compilation fails.

To prevent nested calls, the compiler wrapper removes the elements
of SPACK_ENV_PATH from PATH.

Sadly, the autotest framework appends a slash to each element
of PATH when adding AUTOTEST_PATH to the PATH for the tests,
and some tests like those of GNU bison run cc inside the test.

Thus, ensure that PATH cleanup works even with trailing slashes.

This fixes the autotest suite of bison, compiling hundreds of
bison-generated test cases in a autotest-generated testsuite.

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>
